core/src/main/java/io/temporal/samples/hello/HelloUpdate.java

```diff
@@ -0,0 +1,287 @@
+ *  use this file except in compliance with the License. A copy of the License is
+ *  located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ *  or in the "license" file accompanying this file. This file is distributed on
+ *  an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ *  express or implied. See the License for the specific language governing
+ *  permissions and limitations under the License.
+ */
+
+package io.temporal.samples.hello;
+
+import com.google.common.base.Throwables;
+import io.temporal.activity.ActivityOptions;
+import io.temporal.client.WorkflowClient;
+import io.temporal.client.WorkflowOptions;
+import io.temporal.client.WorkflowStub;
+import io.temporal.client.WorkflowUpdateException;
+import io.temporal.failure.ApplicationFailure;
+import io.temporal.serviceclient.WorkflowServiceStubs;
+import io.temporal.worker.Worker;
+import io.temporal.worker.WorkerFactory;
+import io.temporal.workflow.SignalMethod;
+import io.temporal.workflow.UpdateMethod;
+import io.temporal.workflow.UpdateValidatorMethod;
+import io.temporal.workflow.Workflow;
+import io.temporal.workflow.WorkflowInterface;
+import io.temporal.workflow.WorkflowMethod;
+import java.time.Duration;
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * Sample Temporal workflow that demonstrates how to use workflow update methods to update a
+ * workflow execution from external sources.
+ *
+ * <p>Note: Make sure to set {@code frontend.enableUpdateWorkflowExecution=true} in your Temporal
+ * config to enabled update.
+ */
+public class HelloUpdate {
+
+  // Define the task queue name
+  static final String TASK_QUEUE = "HelloUpdateTaskQueue";
+
+  // Define the workflow unique id
+  static final String WORKFLOW_ID = "HelloUpdateWorkflow";
+
+  /**
+   * The Workflow Definition's Interface must contain one method annotated with @WorkflowMethod.
+   *
+   * <p>Workflow Definitions should not contain any heavyweight computations, non-deterministic
+   * code, network calls, database operations, etc. Those things should be handled by the
+   * Activities.
+   *
+   * @see WorkflowInterface
+   * @see WorkflowMethod
+   */
+  @WorkflowInterface
+  public interface GreetingWorkflow {
+    /**
+     * This is the method that is executed when the Workflow Execution is started. The Workflow
+     * Execution completes when this method finishes execution.
+     */
+    @WorkflowMethod
+    List<String> getGreetings();
+
+    /*
+     * Define the workflow addGreeting update method. This method is executed when the workflow
+     * receives an update request.
+     */
+    @UpdateMethod
+    int addGreeting(String name);
+
+    /*
+     * Define an optional workflow update validator. The validator must take the same parameters as the update handle.
+     * The validator is run before the update handle.
+     * If the validator fails by throwing any exception the update request will be rejected and the handle will not run.
```

Comment:
I would mention that the update is not recorded in the history if thrown from the validator and is recorded if thrown from the update method itself.

Reply:
Also, we need to document that validator must never change the workflow state (the same as query method).

Reply:
Yes, both of those are documented in this sample